### DISCUSSION TITLE
Heading Level

### DISCUSSION THREAD

**Discussion by Markusimus** [IBM/ELM-Python-Client#6] [Q&A] [answered]
Hello, 
thanks for developing this tool! It is useful for post processing of the DNG documents.
The only problem I have is with Heading Level. I'm able to identify that the object is Heading ('oslc:instanceShape': 'Heading') but I don't have any information about the level, so the structure of the module can not be restored.
Is there a way how to get level of the Heading?

Thanks,
Marek

**Comment by barny** [comment, marked as answer]
Hi Marek

> thanks for developing this tool! It is useful for post processing of the DNG documents.

Thanks

> Is there a way how to get level of the Heading?

The heading level isn't stored anywhere, I imagine this is because if it were saved in artifacts it would have to be updated every time something was added/deleted/moved.

In the DOORS Next UI, the 'section' attribute shows the heading level and I guess it's dynamically generated.

Have a look at the code in [dn_simple_modulestructure.py](https://github.com/IBM/ELM-Python-Client/blob/master/elmclient/examples/dn_simple_modulestructure.py) there's example of how to work out the section value by scanning the structure - note you'll have to check this code works as well as you need because I've not done an exhaustive test on it, but it does seem to generate sensible heading and section numbers. This code uses the [Module API](https://jazz.net/wiki/bin/view/Main/DNGModuleAPI). 

Ian

**Reply by Markusimus** [reply]
The example in dn_simple_modulestructure.py is exactly what I need.

Thank You!

**Reply by barny** [reply]
FYI It's also possible to get the module structure using the DNG Reportable REST API (not OSLC) which doesn't use/need headers; you can put all the parameters in the URL and also you can access it using a logged-in browser which can be very convenient to test the URL.

For example the following query will return the content of the module, with each value showing the section number of the artifact:

https://SERVER:PORT/rm/publish/modules?moduleUri=MD_COUvD0B5Eeuh3Iiax2L3Ow&oslc_config.context=https%3A%2F%2FSERVER%3APORT%2Frm%2Fcm%2Fstream%2F_B7GQsEB5Eeuh3Iiax2L3Ow
